### PR TITLE
Disable pin menu button in VideoGallery when max pinned participants is reached

### PR DESCRIPTION
**Pull request by mgamis-msft**
# What

- Added `maxPinnedRemoteVideoTiles` prop to VideoGallery.
- Disabling pin menu button on remote VideoTile of VideoGallery when max pinned remote video tiles is reached
- Added enzyme tests for pinned participants enzyme tests including those that test prop `maxPinnedRemoteVideoTiles`
- Added 1 hermetic playwright tests to test that CallComposite will limit max pinned remote video tiles to 4
- Added a revision to API View for Pinned Participants here: https://apiview.dev/Assemblies/Review/7276a0d82fce4755b833fc91def19dbc/8a35753f996445c28156396635624f28?diffRevisionId=80605558e3ea4faa8acc2db2edb2e3fa&doc=False&diffOnly=True

# Why

https://skype.visualstudio.com/SPOOL/_workitems/edit/3043119
https://skype.visualstudio.com/SPOOL/_workitems/edit/3050025
https://skype.visualstudio.com/SPOOL/_workitems/edit/3045927
https://skype.visualstudio.com/SPOOL/_workitems/edit/3050071

# How Tested

Local calling sample, storybook, and through enzyme tests and 1 playwright test

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

**Comment by JamesBurnside** [comment]
LGTM! Just waiting for green CI before approving. Added update_snapshots label to get snapshots of the new e2e test

File: packages/react-composites/tests/browser/call/hermetic/VideoGallery.test.ts
Comment (JamesBurnside) [reply packages/react-composites/tests/browser/call/hermetic/VideoGallery.test.ts:91]: Looks like this is also causing the update snapshots to fail:
image

Comment (mgamis-msft) [reply packages/react-composites/tests/browser/call/hermetic/VideoGallery.test.ts]: I am trying to click the last instance of the more options button which is presumably in an unpinned remote video tile but for some reason this does not work in CI. 
Running this test locally I am able to get this snapshot.
disabled-pin-menu-button-win32

File: packages/communication-react/review/beta/communication-react.api.md
Comment (anjulgarg) [inline packages/communication-react/review/beta/communication-react.api.md:3214]: I am not sure if we ever discussed this prop as part of the public API changes.
Do we really need to introduce this?
I was under the impression that VideoGallery was going to enforce default of 4

Comment (JamesBurnside) [reply packages/communication-react/review/beta/communication-react.api.md:3214]: If its helpful I really like this prop! In there a reason to limit to 4?

**Comment by JamesBurnside** [comment]
I actually worry about disabling the button - it's not clear to the user _why_ it's disabled and how they can go about fixing that, can you check in with design on this?

**Comment by JamesBurnside** [comment]
nit: the pin icon doesn't quite have the correct padding around it, the text is closer to the pin than the pin is to the left hand side:
image

File: packages/communication-react/review/beta/communication-react.api.md
Comment (mgamis-msft) [reply packages/communication-react/review/beta/communication-react.api.md:3191]: I can hardcode max limit of 4 pinned remote video tiles to disable the pin menu item just to avoid the API change. And we can add this prop in the future from customer signals by reverting the commit hash on this branch.

File: packages/react-composites/tests/browser/call/hermetic/VideoGallery.test.ts
Comment (mgamis-msft) [reply packages/react-composites/tests/browser/call/hermetic/VideoGallery.test.ts]: Issue fixed. This fails on mobile UI tests because ellipses button is not shown on mobile. Fixed.

**Comment by mgamis-msft** [comment]
> nit: the pin icon doesn't quite have the correct padding around it, the text is closer to the pin than the pin is to the left hand side: image
I think the snapshot you saw might have been outdated because the margins look fine to me.
image
I looked at the margins on storybook and they seem ok as well.
image

**Comment by JamesBurnside** [comment]
Going by the [latest snapshot update](https://github.com/Azure/communication-ui-library/pull/2610/commits/eb94476cf0be744c052acabc982bbc9a27368d88) the horizontal margins are still a little off (by ~30% depending on resolution), and the vertical direction isn't center aligned:
image
^ The icon should be vertically centered with the text and not bottom-aligned?

I see the same issue in storybook:
image


For reference this is what Teams' looks like:
image

Icon alignment is tricky (fluent icons aren't the best). Happy to defer to design and have them sign off or not on if this is an acceptable level of alignment

**Comment by mgamis-msft** [comment]
> Going by the [latest snapshot update](https://github.com/Azure/communication-ui-library/pull/2610/commits/eb94476cf0be744c052acabc982bbc9a27368d88) the horizontal margins are still a little off (by ~30% depending on resolution), and the vertical direction isn't center aligned: image ^ The icon should be vertically centered with the text and not bottom-aligned?
> 
> I see the same issue in storybook: image
> 
> For reference this is what Teams' looks like (notice the pin icon extends ~the same amount above and below the P in Pin): image
> 
> Icon alignment is tricky (fluent icons aren't the best). Happy to defer to design and have them sign off or not on if this is an acceptable level of alignment

I was able to change the icon size so that the icon is vertically aligned with the menu text. But there is now a little more space on the right side due that natural padding of the span element which is what is the menu text is.
image
image
image. Let me know if I should file a bug to fix this. Thanks.

**Comment by mgamis-msft** [comment]
> I actually worry about disabling the button - it's not clear to the user _why_ it's disabled and how they can go about fixing that, can you check in with design on this?

Talked with Alex P and we are now indicating that the limit is reached. Please see updated snapshot.

**Comment by JamesBurnside** [comment]
>  was able to change the icon size so that the icon is vertically aligned with the menu text

The icon in the snapshot is still not vertically centered:
image

File: packages/react-components/src/localization/locales/en-US/strings.json
Comment (JamesBurnside) [inline packages/react-components/src/localization/locales/en-US/strings.json:177]: nit: remove the `when` here, just `pinParticipantForMeLimitReached`

**Review by JamesBurnside — APPROVED** [review]

File: change-beta/@azure-communication-react-12a92b59-bacd-414a-927d-fe69b009f6cf.json
Comment (prabhjot-msft) [inline change-beta/@azure-communication-react-12a92b59-bacd-414a-927d-fe69b009f6cf.json:3]: As we are not adding the `maxPinnedRemoteVideoTiles ` prop, the comment seems misleading.

File: change/@azure-communication-react-3bfec36c-0908-4eb5-b99a-f83fdc6d7f26.json
Comment (prabhjot-msft) [inline change/@azure-communication-react-3bfec36c-0908-4eb5-b99a-f83fdc6d7f26.json:3]: same as above

File: packages/react-composites/tests/browser/call/hermetic/VideoGallery.test.ts
Comment (prabhjot-msft) [inline packages/react-composites/tests/browser/call/hermetic/VideoGallery.test.ts:106]: we can use `waitforselector` helper method to avoid this repetitive code as it takes care of screenshot on failure

File: packages/react-composites/tests/browser/call/hermetic/VideoGallery.test.ts
Comment (prabhjot-msft) [inline packages/react-composites/tests/browser/call/hermetic/VideoGallery.test.ts:112]: same as above

**Review by prabhjot-msft — APPROVED** [review]

**Pull request merged**
